### PR TITLE
[mlir][GPU] Do not strip location info when lowering to NVVM

### PR DESCRIPTION
**Pull request by matthias-springer**
This is needed for a subsequent commit that reads location information when lowering `gpu.assert`.

### CODE REVIEW

File: mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp [hunk 1]
```diff
@@ -73,7 +73,6 @@ void buildCommonPassPipeline(
 //===----------------------------------------------------------------------===//
 void buildGpuPassPipeline(OpPassManager &pm,
                           const mlir::gpu::GPUToNVVMPipelineOptions &options) {
-  pm.addNestedPass<gpu::GPUModuleOp>(createStripDebugInfoPass());
```
Comment (joker-eph) [inline mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp:76]: Why was this added in the first place?

Comment (matthias-springer) [reply mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp:76]: Not sure. Looking at the git history (https://reviews.llvm.org/D155463), there was a test case that had it in the pass pipeline. See also https://reviews.llvm.org/D103187. Maybe @ThomasRaoux knows.

**Review by joker-eph — APPROVED** [review]

File: mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp (hunk 1 above)
Comment (ThomasRaoux) [reply mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp:76]: I remember that passing ptx with debug info used to crash nvidia driver. I had not debugged it at the time though

Comment (matthias-springer) [reply mlir/lib/Dialect/GPU/Pipelines/GPUToNVVMPipeline.cpp:76]: Good to know. Given that the `printf` integration test works on my machine, this seems to be resolved.
